src/GuiHandler.java

```diff
@@ -0,0 +1,373 @@
+        guesWordLenght.setVisible(true);
+        
+        setCorrectLettersLength();
+        
+        lblSetAnswer.setVisible(false);
+        txtSetAnswer.setVisible(false);
+        btnSetAnswer.setVisible(false);
+        
+        txtSetAnswer.clear();
+    }
+    
+    public void checkLetter()
+    {
+    	if(gameRunningStatus == false) {errorPopupHandler(0, "", "De game is niet meer actief. \nVoer een nieuw woord in om de game te starten."); return;}
+    	if(txtTryALetter.getText().matches(".*\\d.*")) {errorPopupHandler(0, "", "Het woord mag alleen letters bevatten"); return;}
+    	if(answer.length() <= 0) {errorPopupHandler(0, "", "Voer een woord in om te raden"); return;}
+    	
+        String letter = txtTryALetter.getText();
+        int inputLenght = letter.length();
+        
+        try 
+        {
+        	if(inputLenght > 1 || inputLenght <= 0) throw new IllegalArgumentException();
+        	
+        	if(answer.indexOf(letter) < 0) {wrongGuesHandler(letter); return;}
+        	correctGuesHandler(letter);
+        }
+        catch(IllegalArgumentException ilex)
+        {
+        	errorPopupHandler(inputLenght, "Voer maximaal 1 letter in", "Voer mininaal 1 letter in");
+        }
+    }
+
+    public void checkWord()
+    {
+    	if(gameRunningStatus == false) {errorPopupHandler(0, "", "De game is niet meer actief. \nVoer een nieuw woord in om de game te starten."); return;}
+    	if(txtTryAWord.getText().matches(".*\\d.*")) {errorPopupHandler(0, "", "Het woord mag alleen letters bevatten"); return;}
+    	if(answer.length() <= 0) {errorPopupHandler(0, "", "Voer een woord in om te raden"); return;}
+
+    	String word = txtTryAWord.getText();
+        int inputLenght = word.length();
+        
+        try 
+        {
+        	if(inputLenght < 3) throw new IllegalArgumentException();
+        	
+        	if(!word.equalsIgnoreCase(answer)) {loseGameHandler(""); return;}
+        	winGameHandler();
+        }
+        catch(IllegalArgumentException ilex)
+        {
+        	errorPopupHandler(inputLenght, "Voer minimaal 1 woord in van 3 letters", "Voer minimaal 1 woord in van 3 letters");
+        }
+    }
+    
+    //Helper functions
+    private void correctGuesHandler(String letter)
+    {
+    	StringBuilder fillCorrectLetter = new StringBuilder(correctLetters);
+    	char correctCharLetter = letter.charAt(0);
+    	
+    	long amountCharInString = answer.length() - (correctLetters.chars().filter(ch -> ch != correctCharLetter).count());
+    	int correctLetterIndex = answer.indexOf(letter);
+
+    	if(correctLetters.indexOf(letter) >= 0)
+    	{
+    		
+    		for(int i = 0; i < amountCharInString + 1; i++)
+        	{
+        		if(i == 0)
+        		{
+        			correctLetterIndex = answer.indexOf(letter);
+        		}
+        		else if(i > 0)
+        		{
+        			correctLetterIndex = answer.indexOf(letter, answer.indexOf(letter) + i);
+        		}
```

Comment:
is die 0 check nodig? er is geen verschil naast second argument ik zie niet zo snel wat je hier doet. 
hier kan je zelf kijken of het "netter" kan